idaes/generic_models/properties/core/eos/enrtl.py

```diff
@@ -109,70 +158,171 @@ def rule_X(b, j):
+                    sum(b.mole_frac_phase_comp_true[pname, s]
+                        for s in molecular_set))
+        b.add_component(pname+"_vol_mol_solvent",
+                        Expression(rule=rule_vol_mol_solvent,
+                                   doc="Mean molar volume of solvent"))
+
+        # Mean relative permitivity of solvent
+        def rule_eps_solvent(b):  # Eqn 78
+            return (sum(b.mole_frac_phase_comp_true[pname, s] *
+                        get_method(b, "relative_permittivity_liq_comp", s)(
+                            b, cobj(b, s), b.temperature) *
+                        b.params.get_component(s).mw
+                        for s in molecular_set) /
+                    sum(b.mole_frac_phase_comp_true[pname, s] *
+                        b.params.get_component(s).mw
+                        for s in molecular_set))
+        b.add_component(pname+"_relative_permittivity_solvent",
+                        Expression(
+                            rule=rule_eps_solvent,
+                            doc="Mean relative permittivity  of solvent"))
+
+        # Debye-Huckel parameter
+        def rule_A_DH(b):  # Eqn 61
+            # Note: Where the paper refers to the dielectric constant, it
+            # actually means the electric permittivity of the solver
+            # eps = eps_r*eps_0 (units F/m)
+            v = pyunits.convert(getattr(b, pname+"_vol_mol_solvent"),
+                                pyunits.m**3/pyunits.mol)
+            eps = getattr(b, pname+"_relative_permittivity_solvent")
+            eps0 = Constants.vacuum_electric_permittivity
+            return ((1/3)*(2*Constants.pi*Constants.avogadro_number/v)**0.5 *
+                    (Constants.elemental_charge**2/(
+                        eps*eps0*Constants.boltzmann_constant *
+                        b.temperature))**(3/2)
+                    )
+        b.add_component(pname+"_A_DH",
+                        Expression(
+                            rule=rule_A_DH,
+                            doc="Debye-Huckel parameter"))
+
+        # Long-range (PDH) contribution to activity coefficient
+        def rule_log_gamma_pdh(b, j):
+            A = getattr(b, pname+"_A_DH")
+            I = getattr(b, pname+"_ionic_strength")
+            I0 = getattr(b, pname+"_ionic_strength_ref")
+            rho = ClosestApproach
+            if j in molecular_set:
+                # Eqn 69
+                # Note typo in original paper. Correct power for I is (3/2)
+                return (2*A*I**(3/2)/(1+rho*I**(1/2)))
+            elif j in b.params.ion_set:
+                # Eqn 70
+                z = cobj(b, j).config.charge
+                return (-A*((2*z**2/rho) *
+                            log((1+rho*I**0.5)/(1+rho*I0**0.5)) +
+                            (z**2*I**0.5 - 2*I**(3/2)) / (1+rho*I**0.5) -
+                            (2*I*I0**-0.5) / (1+rho*I0**0.5) *
+                            ref_state.ndIdn(b, pname, j)))
```

Comment:
You may have noted this somewhere already, but a `#TODO` note for later on -- the last term will go to 0 since, `ndIdn=0` for the unsymmetric reference state (according to eqn 71).

Reply:
The unsymmetric reference case will handle that when implemented by making `ndIdn = 0`, at which point Pyomo is smart will drop the entire term (it is smart enough to process out any term which is multiplied by 0).